src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -10,14 +11,18 @@ interface ProductShelfProps {
 }
 
 function ProductShelf({ products }: ProductShelfProps) {
+  const haveProducts = products && products?.length > 0
```

Comment:
```suggestion
  const haveProducts = products?.length > 0
```

maybe we can put it inline on line 17